src/pooling.jl

```diff
@@ -154,19 +168,69 @@ end
 
 
 """
-    meanpool(x, k::NTuple; pad=0, stride=k)
+    meanpool(x, k::NTuple{N, Integer}; pad=0, stride=k)
 
 Perform mean pool operation with window size `k` on input tensor `x`.
+
+* `x` and `k`: Usually, ndim(x) ∈ [3, 5], length(k) ∈ [1, 3], s.t. ndim(x) == length(k) + 2
+* `pad`: See [`pad_zeros`](@ref) for details.
+* `stride`: Stride for each spatial axis. `k` as default if not present.
 """
 function meanpool(x, k::NTuple{N, Integer}; pad=0, stride=k) where N
+    ndims(x) == length(k) + 2 || error("meanpool expects ndims(x) == length(k)+2,
+                                        dimension of x is $(ndims(x)),
+                                        length of k need $(ndims(x) - 2),
+                                        but now it's $(length(k))")
     pad = expand(Val(N), pad)
     stride = expand(Val(N), stride)
     pdims = PoolDims(x, k; padding=pad, stride=stride)
     return meanpool(x, pdims)
 end
 
 
-for pool in [:maxpool, :meanpool]
+"""
+    lppool(x, p::Number, k::NTuple{N, Integer}; pad=0, stride=k)
+
+Perform Lp pool operation with value of the Lp norm `p` and `window size `k` on input tensor `x`.
+
+* `x` and `k`: Usually, ndim(x) ∈ [3, 5], length(k) ∈ [1, 3], s.t. ndim(x) == length(k) + 2
+* `pad`: See [`pad_zeros`](@ref) for details.
+* `stride`: Stride for each spatial axis. `k` as default if not present.
+
+For each element `x` in (k × k) window, lppool computes `(∑ x^p)^(1 / p)` as output.
+
+* When p = 1, lppool(x, p, k) ./ prod(k) ≈ meanpool(x, k)
+* When p = 2, lppool(x, p, k).^2 ./ prod(k) ≈ meanpool(x.^2, k)
+
+!!! warning
+
+    Theoretically, when `p -> ∞`, lppool(x, p, k) ≈ maxpool(x, k).
+    But it's not correct in julia. Given an arbitrary Number `n`,
+    ```jldoctest
+    julia> n = 10
+    10
+
+    julia> ans^Inf
+    Inf
+
+    julia> ans^(1/Inf)
+    1.0
```

Comment:
This isn't the way. It's not a warning box. And "it's not correct in julia" is simply false. 

Conceptually this function is like `norm(x,p)` on a moving window. The least surprising option is that it should accept all the same `p` values which `norm` does. 

There's a mathematically sensible meaning for when p becomes infinite, etc. But the implementation of `LinearAlgebra.norm`  doesn't fall out of floating point arithmetic like this comment suggest it should. It branches on `if p == -Inf` etc. 

We could duplicate all those branches here. And test (like LinearAlgebra does) that they all work. 

It's also fine to decide that nobody needs `lppool(x, -3.3)`. In that case, the two things needed here are (1) a comment like "accepts `0 < p < Inf`", no more, and (2) an error on `p` outside the accepted range. But no lesson on IEEE floats, no giant warning box.